### DISCUSSION TITLE
Editor feature requests / notes

### DISCUSSION THREAD

**Discussion by infovore** [shorepine/tulipcc#292] [Ideas]
Currently hacking away on Tulip Desktop, and have been using the editor enough to have some vague notes, that I'm not really ready to convert to issues:

- `ctrl+r` to open a file seemed - I need to check - to overwrite an existing file to blank when I used it, which was annoying, as now I have to rewrite a bunch of things...
- syntax highlighting is a bit rough-and-ready, especially with things like variables containing numbers, or around single quotes.
- would dearly like some more keybinds for by-word/by-line. (I mean, I'd _really_ like vim bindings, but this is not an editor project)
- some form of copy/paste, even if it's just repeat line (back to vim again) would be useful.

Also, it'd be awesome if Tulip Desktop somehow could support clipboard...

**Comment by bwhitman** [comment]
the last one -- control-K does "yank" / control-U "unyank" like `nano` 
The rest, yes. The editor needs some love. I was hoping to write it in Python or borrow an implementation and port it to the tulip TFB / "terminal"  like https://github.com/prompt-toolkit/pyvim

**Reply by infovore** [reply]
Thanks for the yank/unyank reminder!
